### PR TITLE
refactor: Render function size reduce

### PR DESCRIPTION
**Pull request by saxenanihal95**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

1. Describe the source of requirement, like related issue link.

Optizime code according to codebeat : [#15288](https://github.com/ant-design/ant-design/issues/15288)

2. Describe the problem and the scenario.

render code of list.tsx is too long.

### 💡 Solution

Moved some of the render code to functions.

### 📝 Changelog

- English Changelog:
- Chinese Changelog (optional):

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/transfer/list.tsx [hunk 1]
```diff
@@ -91,6 +91,99 @@ export default class TransferList extends React.Component<TransferListProps, any
...
+          disabled={disabled}
+        />
+      </div>
+    );
+  }
+
+  getListBody(
+    prefixCls: string,
+    onScroll: Function,
+    search: any,
```
Comment (DiamondYuan) [inline components/transfer/list.tsx]: 我觉得尽量不要用 any 比较好。

Comment (saxenanihal95) [reply components/transfer/list.tsx]: Ok I will check the type of it and add in place of any

**Comment by afc163** [comment]
You may wait for https://github.com/ant-design/ant-design/pull/16026 otherwise there would be a lot of conflicts.

**Comment by saxenanihal95** [comment]
Ok will wait for that to merge

**Comment by afc163** [comment]
You can restart your job now.

**Comment by afc163** [comment]
This pull request **introduces 1 alert** when merging 4e9b3d5379bea53906387cec5acdad0a35050ea0 into 34d8ff8eb49f6c88827ae0b0fc72ebb04bba8196 - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-322e593d7d73358922075888dd376650289741a8)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by saxenanihal95** [comment]
yes fixed that 

> This pull request **introduces 1 alert** when merging [4e9b3d5](https://github.com/ant-design/ant-design/commit/4e9b3d5379bea53906387cec5acdad0a35050ea0) into [34d8ff8](https://github.com/ant-design/ant-design/commit/34d8ff8eb49f6c88827ae0b0fc72ebb04bba8196) - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-322e593d7d73358922075888dd376650289741a8)
> 
> **new alerts:**
> 
> * 1 for Unused variable, import, function or class
> 
> _Comment posted by [LGTM.com](https://lgtm.com)_

**Pull request merged**